clab/exec/exec.go

```diff
@@ -51,23 +51,13 @@ func NewExecCmdFromSlice(cmd []string) *ExecCmd {
 	}
 }
 
-type ExecResultHolder interface {
-	GetStdOutString() string
-	GetStdErrString() string
-	GetStdOutByteSlice() []byte
-	GetStdErrByteSlice() []byte
-	GetReturnCode() int
-	GetCmdString() string
-	Dump(format string) (string, error)
-	String() string
-}
-
 // ExecResult represents a result of a command execution.
 type ExecResult struct {
-	Cmd        []string `json:"cmd"`
-	ReturnCode int      `json:"return-code"`
-	Stdout     string   `json:"stdout"`
-	Stderr     string   `json:"stderr"`
+	Cmd        []string        `json:"cmd"`
+	ReturnCode int             `json:"return-code"`
+	Stdout     string          `json:"stdout"`
+	Json       json.RawMessage `json:"json"`
```

Comment:
@steiler I haven't been able to find a way to keep just `stdout` field playing nicely both with underlying valid json and just strings simultaneously.

Maybe some empty interface type for stdout is needed with dynamic type assertions, but I failed to come up with a solution to that.

Therefore, two distinct fields:

- stdout - for string based outputs
- json - for valid json outputs

Reply:
discard, I think I found a way